skimage/_shared/fast_exp.h

```diff
@@ -38,3 +41,36 @@ __inline double fast_exp (double y)
 
     return _eco.d;
 }
+
+
+#define EXPF_A (184.6649652337873)  /* 2^7÷ln2 */
+/* For min. RMS error */
+#define EXPF_BC 16249              /* 127*2^7 - 7 */
+
+
+__inline float fast_expf (float y)
```

Comment:
Is the 32-bit version really needed? When you work with small volumes/images the memory conservation is negligible, and with large ones - double precision is needed.

Reply:
I introduces the 32-bit version because I still don't know what Cython is doing when fast_exp is called with a single precision float. I wanted to make it explicit.

Reply:
I'm not a maintainer of this project, so I can only suggest, but to me it looks like `fast_exp` just shouldn't be called on single precision floats. Because of the reasoning above - with small datasets usage of double precision does not hurt much, and with large datasets - single precision is not enough.

Reply:
From my point of view, the use of single precision arrays is an explicit user choice. Enforcing the use of double precision in this case is not the right option IMHO.

Reply:
Okay, that is a valid point, but in any case, it feels like there should be no two single precision functions. I'd just pick one that works well, and keep it. But again, I'm not a maintainer, so please regard this as a mere opinion.

Reply:
You are absolutely right, and if you have a look at [one](https://github.com/scikit-image/scikit-image/pull/4322#discussion_r353033407) of my previous comments, you can see that this is kind of exploratory and that the question of which version to keep is still open!